### PR TITLE
EVG-7161 use API models in gql

### PR DESCRIPTION
**Pull request by john-m-liu**
- Replace APIString, APITime, and APIDuration with *string, time.Time, and time.Duration respectively. These are the first 3 commits which probably don't require looking at
- Switch the models returned by GQL resolvers to be the REST models
- Change some patches functions in the rest interface to return API models rather than DB models
- Add an integration test for GQL queries. Take a look at the top of graphql/integration_test.go for how it works - my intent was that people can add tests for query logic/resolvers in the future without writing any go
- Add a task for GQL go tests

### CODE REVIEW

File: graphql/testdata/patch1.graphql [hunk 1]
```diff
@@ -0,0 +1,10 @@
+{
+  userPatches(userId: "testuser") {
+    id
+    variants
+    variantsTasks {
+      name
+      tasks
+    }
+  }
```
Comment (tgrander) [inline graphql/testdata/patch1.graphql]: why only testing these fields as opposed to all fields in type `Patch` ?

Comment (SupaJoon) [reply graphql/testdata/patch1.graphql]: this represents a query that could be made from the frontend and we will probably copy/paste query strings from Spruce to test

**Comment by SupaJoon** [comment]
this works well and I was able to test it with `make test-graphql` thanks john!!

File: graphql/testdata/patch1.graphql (hunk 1 above)
Comment (tgrander) [reply graphql/testdata/patch1.graphql]: yes that part is clear. my question is why not include all the Patch fields as opposed to just some fo them

Comment (john-m-liu) [reply graphql/testdata/patch1.graphql]: I picked some arbitrary fields. I changed it to include all fields

File: graphql/integration_test.go [hunk 2]
```diff
@@ -0,0 +1,117 @@
...
+	"go.mongodb.org/mongo-driver/bson"
+	"go.mongodb.org/mongo-driver/mongo"
+)
+
+type graphQLSuite struct {
+	url     string
+	apiUser string
+	apiKey  string
+
+	suite.Suite
```
[the remark this answers is not in the document]
Comment (john-m-liu) [reply graphql/integration_test.go]: I find that it keeps setup/individual tests organized much better

File: makefile [hunk 3]
```diff
@@ -3,7 +3,7 @@ name := evergreen
 buildDir := bin
 tmpDir := $(abspath $(buildDir)/tmp)
 nodeDir := public
-packages := $(name) agent operations cloud command db util plugin units
+packages := $(name) agent operations cloud command db util plugin units, graphql
```
[the remark this answers is not in the document]
Comment (john-m-liu) [reply makefile]: oops, fixed

File: graphql/integration_test.go (hunk 2 above)
[the remark this answers is not in the document]
Comment (tychoish) [reply graphql/integration_test.go]: I'm less sold on this in general, and think you end up writing less flexible code, but it's fine

**Pull request merged**